core/src/main/scala/kafka/zookeeper/ZooKeeperClient.scala

```diff
@@ -47,6 +47,8 @@ class ZooKeeperClient(connectString: String, sessionTimeoutMs: Int, connectionTi
   private val isConnectedOrExpiredCondition = isConnectedOrExpiredLock.newCondition()
   private val zNodeChangeHandlers = new ConcurrentHashMap[String, ZNodeChangeHandler]().asScala
   private val zNodeChildChangeHandlers = new ConcurrentHashMap[String, ZNodeChildChangeHandler]().asScala
+  private val stateChangeHandlers = new ConcurrentHashMap[String, StateChangeHandler]().asScala
```

Comment:
ZooKeeperClient has only a single underlying state, so it doesn't feel right to me that you're allowing multiple state change handlers. Why not just let the one state change handler do multiple things?

Reply:
We are passing single KafkaZKClient instance to multiple components/modules.  So with single state change handler it is bit difficult to invoke/organize different parts of code.  with multiple handlers, each handler can invoke their respective methods. 
 let me know you thoughts.